homeassistant/components/sonos/speaker.py

```diff
@@ -685,7 +699,19 @@ def update_group_for_uid(self, uid: str) -> None:
     @callback
     def async_update_groups(self, event: SonosEvent) -> None:
         """Handle callback for topology change event."""
-        if not hasattr(event, "zone_player_uui_ds_in_group"):
+        if xml := event.variables.get("zone_group_state"):
+            zgs = ET.fromstring(xml)
+            for vanished_device in zgs.find("VanishedDevices"):
+                if (reason := vanished_device.get("Reason")) == "UNKNOWN":
```

Comment:
Should we whitelist instead?  That might be safer

Reply:
Perhaps. So far I've only seen "sleeping" as a valid reason.